app/demo-loop/Runner.cc

```diff
@@ -77,17 +77,18 @@ Runner::Runner(RunnerInput const& inp,
 
     ScopedRootErrorHandler scoped_root_error;
     this->build_core_params(inp, std::move(output));
-    this->build_step_collectors(inp);
-    this->build_transporter_input(inp);
-    this->build_primaries(inp);
-    use_device_ = inp.use_device;
-
-    if (root_manager_)
+    if (!inp.mctruth_filename.empty())
     {
-        // Store input and CoreParams data
+        // Initialize ROOT file; Store input and core params
+        root_manager_
+            = std::make_shared<RootFileManager>(inp.mctruth_filename.c_str());
```

Comment:
I'm thinking about this again... if we're *only* going to create a ROOT manager if we're making an MC truth file, then this should go in the `build_step_collectors`. I don't like having the big block here required to be among the `this->build` list... maybe make a `this->build_root` method...